### PR TITLE
Add `space` option

### LINKED ISSUE #7: Add option to add a space either side of ellipsis
An option to add a space either side of the ellipsis would be nice. thanks

### PR DESCRIPTION
**Pull request by satyarohith**
closes #7

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -73,6 +73,12 @@ Values: `start` `middle` `end`
 
 Position to truncate the string.
 
+##### space
+
+Type: `string`<br>
+Values: `left``right`
```
Comment (sindresorhus) [inline readme.md]: This should just be a boolean, and when true:

- `foo ...`
- `... foo`
- `fo ... oo`

Comment (satyarohith) [reply readme.md]: Should this depend on the `position` option?

Comment (sindresorhus) [reply readme.md]: Yes

File: readme.md [hunk 2]
```diff
@@ -73,6 +73,12 @@ Values: `start` `middle` `end`
 
 Position to truncate the string.
 
+##### space
+
+Type: `boolean`<br>
+Values: `true` `false`
+
+Add space to ellipsis. Depends on the `position` option.
```
Comment (sindresorhus) [inline readme.md]: Add a code example here of how it looks with the different `position` options. It's not immediate clear to the user how it would look.

File: index.js [hunk 3]
```diff
@@ -33,13 +33,37 @@ module.exports = (input, columns, opts) => {
...
+			ellipsis = ' ' + ellipsis;
+		}
+
+		return sliceAnsi(input, 0, columns - (space ? 2 : 1)) + ellipsis;
 	}
 
-	throw new Error(`Expected \`options.position\` to be either \`start\`, \`middle\` or \`end\`, got ${position}`);
+	throw new Error(
+		`Expected \`options.position\` to be either \`start\`, \`middle\` or \`end\`, got ${position}`
+	);
```
Comment (sindresorhus) [inline index.js]: Don't do unrelated changes.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

File: readme.md (hunk 2 above)
Comment (satyarohith) [reply readme.md]: I have added them to the main example.

**Comment by satyarohith** [comment]
@sindresorhus Updated!

**Comment by sindresorhus** [comment]
Regarding https://github.com/sindresorhus/cli-truncate/pull/10/files#r271130173, the main example code block already have too many examples. Add it in the options docs, like I commented.

**Comment by sindresorhus** [comment]
You also need to add it to https://github.com/sindresorhus/cli-truncate/blob/master/index.d.ts

File: index.js [hunk 4]
```diff
@@ -34,16 +34,32 @@ module.exports = (input, columns, options) => {
...
+		if (space === true) {
+			ellipsis = ' ' + ellipsis + ' ';
+		}
+
 		const half = Math.floor(columns / 2);
-		return sliceAnsi(input, 0, half) + ellipsis + sliceAnsi(input, length - (columns - half) + 1, length);
+		return (
+			sliceAnsi(input, 0, half) +
+			ellipsis +
+			sliceAnsi(input, length - (columns - half) + (space ? 3 : 1), length)
```
Comment (sindresorhus) [inline index.js]: Too many magic numbers. `3` and `1` should be variables with a descriptive name. Same in the other places.

File: readme.md [hunk 5]
```diff
@@ -73,6 +83,12 @@ Values: `start` `middle` `end`
 
 Position to truncate the string.
 
+##### space
+
+Type: `boolean`
+
+Add space to ellipsis. Depends on the `position` option.
```
Comment (sindresorhus) [inline readme.md]: This description could be improved. You're not adding space **to** the ellipsis, you're adding space **between** the text and the ellipsis.

**Comment by sindresorhus** [comment]
@ammarbinfaisal Can you help review?

File: test.js [hunk 6]
```diff
@@ -19,4 +19,7 @@ test('main', t => {
 	t.is(cliTruncate('unicorn', 6, {position: 'start'}), '…icorn');
 	t.is(cliTruncate('unicorn', 5, {position: 'middle'}), 'un…rn');
 	t.is(cliTruncate('unicorns', 6, {position: 'middle'}), 'uni…ns');
+	t.is(cliTruncate('unicorns', 5, {position: 'end', space: true}), 'uni …');
```
Comment (sindresorhus) [inline test.js]: These should be in a separate test named `space option`. I'd also like to see many more assertions with different kinds of sentences.

File: index.js (hunk 4 above)
Comment (satyarohith) [reply index.js]: I named it `ellipsisWidth`. :)

File: readme.md (hunk 5 above)
Comment (satyarohith) [reply readme.md]: I updated the description.

File: test.js (hunk 6 above)
Comment (satyarohith) [reply test.js]: I have added a few more tests, let me know if we need more.

File: test.js [hunk 7]
```diff
@@ -20,3 +20,13 @@ test('main', t => {
...
 });
+
+test('space option', t => {
+	t.is(cliTruncate('unicorns', 5, {position: 'end', space: true}), 'uni …');
+	t.is(cliTruncate('unicorns', 6, {position: 'start', space: true}), '… orns');
+	t.is(cliTruncate('unicorns', 7, {position: 'middle', space: true}), 'uni … s');
+	t.is(cliTruncate('unicorns', 5, {position: 'end', space: false}), 'unic…');
+	t.is(cliTruncate('\u001B[31municorn\u001B[39m', 6, {space: true}), '\u001B[31munic\u001B[39m …');
+	t.is(cliTruncate('Plant a tree every day.', 14, {space: true}), 'Plant a tree …');
+	t.is(cliTruncate('안녕하세요', 4, {space: true}), '안 …', 'wide char');
```
Comment (sindresorhus) [inline test.js]: The 3 above tests are good, but you need to duplicate for the `position` options too.

**Comment by satyarohith** [comment]
@sindresorhus Updated!
